bin/proving-service/src/proxy/mod.rs

```diff
@@ -117,13 +116,18 @@ impl LoadBalancerState {
         })
     }
 
-    /// Marks the given worker as available.
+    /// Marks the given worker as available and moves it to the end of the list.
     ///
     /// If the worker is not in the list, it won't be added.
     pub async fn add_available_worker(&self, worker: Worker) {
```

Comment:
Does this need to be public? Also, similar question about `pop_available_worker()`.